lib/react_client/react_interop.dart

```diff
@@ -127,17 +128,156 @@ class JsRef {
+///   setClientConfiguration();
+///   final ref = react.createRef<Element>();
+///
+///   react_dom.render(LogProps({'component': FancyButton, 'ref': ref}),
+///       querySelector('#idOfSomeNodeInTheDom'));
+///
+///   // You can still get a ref directly to the DOM button:
+///   final buttonNode = ref.current;
+/// }
+/// ```
 /// See: <https://reactjs.org/docs/forwarding-refs.html>.
-ReactJsComponentFactoryProxy forwardRef(Function(Map props, Ref ref) wrapperFunction) {
-  var hoc = React.forwardRef(allowInterop((JsMap props, JsRef ref) {
+ReactJsComponentFactoryProxy forwardRef(Function(Map props, Ref ref) wrapperFunction,
+    {String displayName = 'Anonymous'}) {
+  final wrappedComponent = allowInterop((JsMap props, JsRef ref) {
     final dartProps = JsBackedMap.backedBy(props);
     final dartRef = Ref.fromJs(ref);
     return wrapperFunction(dartProps, dartRef);
-  }));
+  });
+  defineProperty(wrappedComponent, 'displayName', jsify({'value': displayName}));
+
+  var hoc = React.forwardRef(wrappedComponent);
 
   return new ReactJsComponentFactoryProxy(hoc, shouldConvertDomProps: false);
 }
 
+/// Shorthand convenience function to wrap a JS component that utilizes `forwardRef`.
+///
+/// __Example:__
+///
+/// ```dart
+/// // Assuming that your app includes the JS file containing the MaterialUI components...
+/// @JS()
+/// class MaterialUI {
+///   external static ReactClass get IconButton;
+/// }
+///
+/// /// This makes it so that the `ref` set on the Dart `IconButton` component
+/// /// gets passed all the way to the root node rendered by the JS `MaterialUI.IconButton` component.
+/// final IconButton = forwardRefToJsComponent(MaterialUI.IconButton);
+///
+/// void main() {
+///   setClientConfiguration();
+///   final iconButtonRootNodeRef = react.createRef<Element>();
+///
+///   react_dom.render(IconButton({'ref': iconButtonRootNodeRef}, someIcon), querySelector('#idOfSomeNodeInTheDom'));
+///
+///   // Prints the HTML rendered by the JS `MaterialUI.IconButton` component.
+///   print(iconButtonRootNodeRef.current.outerHtml);
+/// }
+/// ```
+ReactJsComponentFactoryProxy forwardRefToJs(ReactClass jsClassComponent,
+    {List<String> additionalRefPropKeys = const []}) {
+  return forwardRef((props, ref) {
+    return ReactJsComponentFactoryProxy(jsClassComponent, shouldConvertDomProps: false)({
+      ..._convertNonDefaultRefPropKeysToJs(props, additionalRefPropKeys),
+      'ref': ref,
+    }, props['children']);
+  });
+}
+
+Map _convertNonDefaultRefPropKeysToJs(Map props, List<String> additionalRefPropKeys) {
+  var propsMapToReturn = props;
+  if (additionalRefPropKeys.isNotEmpty) {
+    var propsWithConvertedRefs = {...props};
+    for (var propKey in additionalRefPropKeys) {
+      var ref = propsWithConvertedRefs[propKey];
+
+      // The default 'ref' prop key value gets converted by default
+      // by ReactJsComponentFactoryProxy, so we don't need to do it here.
+      if (propKey != 'ref' && ref is Ref) {
```

Comment:
Should we also have wrapping for callback refs here?

Reply:
@greglittlefield-wf is this what you mean?

```dart
if (ref is Function) {
  propsWithConvertedRefs[propKey] = allowInterop((ReactComponent instance) {
    // Call as dynamic to perform dynamic dispatch, since we can't cast to _CallbackRef<dynamic>,
    // and since calling with non-null values will fail at runtime due to the _CallbackRef<Null> typing.
    return ref(instance?.dartComponent);
  });
} else if (propKey != 'ref' && ref is Ref) {
  propsWithConvertedRefs[propKey] = ref.jsRef;
}
```

Reply:
Yup!

Reply:
Now that I think about it, I almost wonder if we should just make `additionalRefPropKeys` a constructor arg in ReactJsComponentFactoryProxy and have this logic live in `_generateJsProps`...

Reply:
@greglittlefield-wf re: the conversion of ref callbacks... `ReactJsComponentFactoryProxy` calls `_generateJsProps` with `convertCallbackRefValue` set to false.  Should it be true?

Reply:
@aaronlademann-wf No, I think the naming on that is just confusing. That flag controls whether callback refs should be wrapped with a version that looks for the Dart component and tries to pass that in instead, and so should only be necessary for Dart component factories